### PR TITLE
luci-app-wireguard: add description

### PR DESCRIPTION
**Pull request by rwalli**
Add description filed to luci-app-wireguard as in luci-proto-wireguard:
https://github.com/openwrt/luci/commit/140f41b40f54feb4fb8d6ddbbdf3c659c779756d#diff-daba76e1569c21be9540dd6307bd578d

This needs reviewing because I don't know lua.

### CODE REVIEW

**Comment by jow-** [comment]
Please fold all changes into one commit.

File: applications/luci-app-wireguard/luasrc/view/wireguard.htm [hunk 1]
```diff
@@ -179,10 +179,26 @@ <h2>WireGuard Status</h2>
...
     </div>
   <%-
+  local cur = uci.cursor();
   for pkey, peer in pairs(iface.peers) do
+    local desc, tmp_desc, pub_key = "", "", ""
+    cur:foreach("network", "wireguard_" .. ikey, function(s)
+      local tmp_desc, pub_key = "", ""
+      for key, value in pairs(s) do
+        if key == "description" then
+          tmp_desc = value;
```
Comment (jow-) [inline applications/luci-app-wireguard/luasrc/view/wireguard.htm]: No semicolons please

File: applications/luci-app-wireguard/luasrc/view/wireguard.htm [hunk 2]
```diff
@@ -179,10 +179,26 @@ <h2>WireGuard Status</h2>
       </div>
     </div>
   <%-
+  local cur = uci.cursor();
   for pkey, peer in pairs(iface.peers) do
+    local desc, tmp_desc, pub_key = "", "", ""
+    cur:foreach("network", "wireguard_" .. ikey, function(s)
+      local tmp_desc, pub_key = "", ""
+      for key, value in pairs(s) do
```
Comment (jow-) [inline applications/luci-app-wireguard/luasrc/view/wireguard.htm]: Please add `local key, value`

File: applications/luci-app-wireguard/luasrc/view/wireguard.htm [hunk 3]
```diff
@@ -179,10 +179,26 @@ <h2>WireGuard Status</h2>
...
+        end
+        if pub_key == peer.public_key and tmp_desc ~= "" then
+          desc = ': ' .. tmp_desc;
+        end   
+      end
+    end)
     -%>
     <div class="tr">
-      <div class="td" width="33%" style="vertical-align:top"><%:Peer%></div>
+      <div class="td" width="33%" style="vertical-align:top"><%:Peer%><%=desc%></div>
```
Comment (jow-) [inline applications/luci-app-wireguard/luasrc/view/wireguard.htm]: Change this into `<%=pcdata(desc)%>` to avoid XSS

**Comment by rwalli** [comment]
Thank you for your suggestions!

I changed your suggestions locally and squashed all 4 commits into one.
Unfortunately something went wrong and I could not push my changes.

Sorry for inconveniences!

What should I do now?
close this PR?

**Comment by hnyman** [comment]
> squashed all 4 commits into one.
> Unfortunately something went wrong and I could not push my changes.

Note that you need to use "git push -f" to force push this PR feature branch (patch-1). Your new squashed commit will overwrite the original ones at Github, so you need to force it.

See. e.g. https://github.com/openwrt/luci/blob/master/CONTRIBUTING.md#advice-on-pull-requests

**Comment by rwalli** [comment]
I've read several "howto's" including the link you suggested and could not push my local repo.
I really tried many hours, Sorry!

**Comment by jow-** [comment]
Superseded by 0fe2e72f7

**Pull request closed without merging**